spec/models/course_spec.rb

```diff
@@ -819,6 +819,28 @@
         end
       end
     end
+
+    describe ".with_provider_name" do
+      let(:provider) { create(:provider, provider_name: "ACME") }
+      let(:course) { create(:course, provider: provider) }
+      let(:provider_name) { "ACME" }
+
+      before do
+        provider
+        course
+        provider_name
+      end
+
+      context "with ACME provider name" do
+        subject { described_class.with_provider_name(provider_name) }
+        it { is_expected.to contain_exactly(course) }
+      end
+
+      context "with Dave provider name" do
```

Comment:
```suggestion
      context "with the name of a provider that doesn't exist" do
```

Reply:
This would be a separate test, as I'm not explicitly testing for a provider that doesn't exist - there isn't going to be a scenario where the api is looking for that, as users are only able to pick from a valid list of providers. I can rewrite the test to include a provider that doesn't have any courses though?

Reply:
I've now modified the test so that it's checking an existing provider with no courses, and an existing provider with courses. This more accurately reflects the actual situations in which it'll be invoked!